applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+'require form';
+'require fs';
+'require view';
+
+return view.extend({
+    load: function () {
+        return Promise.all([
+            L.resolveDefault(fs.stat('/usr/sbin/nginx'), null),
+            L.resolveDefault(fs.stat('/usr/sbin/uhttpd'), null)
+        ]);
+    },
+
+    render: function (stats) {
+        var nginx_presence = stats[0];
+        var uhttpd_presence = stats[1];
+
+        var m = new form.Map("acme", _("ACME certificates"),
+            _("This configures ACME (Letsencrypt) automatic certificate installation. " +
+                "Simply fill out this to have the router configured with Letsencrypt-issued " +
+                "certificates for the web interface. " +
+                "Note that the domain names in the certificate must already be configured to " +
+                "point at the router's public IP address. " +
+                "Once configured, issuing certificates can take a while. " +
+                "Check the logs for progress and any errors."));
+
+        var s = m.section(form.TypedSection, "acme", _("ACME global config"));
+        s.anonymous = true
+
+        var st = s.option(form.Value, "state_dir", _("State directory"),
+            _("Where certs and other state files are kept."));
+        st.rmempty = false;
+        st.datatype = "directory";
+
+        var ae = s.option(form.Value, "account_email", _("Account email"),
+            _("Email address to associate with account key."))
+        ae.rmempty = false;
+        ae.datatype = "minlength(1)";
```

Comment:
Is this the right datatype?

Reply:
Here a real email address is required here. I cannot find the appropriate datatype from the document. Type `"minlength(1)"` is copied from the original lua source.